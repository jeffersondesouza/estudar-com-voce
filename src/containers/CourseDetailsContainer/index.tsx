import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import './style/style.scss';

import CourseMidleware from '../../store/modules/course/middleware';
import { selectCourse } from '../../store/modules/course/selectors';
import { Course } from '../../models';


export interface IProps {
  dispatchLoad(id: number): void,
  match: any,
  selectedCourse: Course | null,
  isLoadingCourse: boolean,
}

export class CourseDetailsContainer extends React.Component<IProps> {

  componentDidMount() {

    this.props.dispatchLoad(+this.props.match.params.id);
  }

  render() {
    const { selectedCourse, isLoadingCourse } = this.props;
    
    return (
      <div className="details">
        <div>
          <Link to="/cocurses" className="details__back">&larr;</Link>
        </div>
        {(!isLoadingCourse && selectedCourse) &&
          <div>
            <h2 className="title details__title">
              {selectedCourse.name}
              {selectedCourse.highlight && <span className="details__highlight">(novo)</span>}
            </h2>
            <p className="details__description">{selectedCourse.description}</p>
          </div>
        }
      </div>
    )
  }
}

const mapStateToProps = state => ({
  ...state.course,
});

const mapDispatchToProps = dispatch => ({
  dispatchLoad: (id: number) => dispatch(CourseMidleware.loadCourse(id)),
});



export default connect(mapStateToProps, mapDispatchToProps)(CourseDetailsContainer);

const MOCK = {
  "id": 1,
  "name": "Física II",
  "description": "Curso remasterizado e criado inteiramente baseado na sua graduação da POLI USP. Provas passadas resolvidas, teoria explicadinha e muitos exercícios extra. Aproveite!",
  "generic": false,
  "highlight": true
}
