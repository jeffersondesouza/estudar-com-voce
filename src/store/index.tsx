/**
 * Store configuration - Here we configure the application store, making it avalable to the projt by simporting this module 
 */
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from './modules/index';

export const configure = (initialState = {}) => (
    createStore(
        reducer,
        initialState,
        compose(
            applyMiddleware(thunk),
        )
    )
);

export default configure();
