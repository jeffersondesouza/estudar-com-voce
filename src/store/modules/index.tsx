import { combineReducers } from 'redux';

import  courseReducer from './course/reducer';

const rootReducer = combineReducers({
  course:  courseReducer,
});

export default rootReducer;
