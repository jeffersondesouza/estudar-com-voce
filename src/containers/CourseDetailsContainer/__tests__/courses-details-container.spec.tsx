import * as React from 'react';
import axios from 'axios';

import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';

import * as renderer from 'react-test-renderer';

import store from '../../../store';
import CourseDetailsContainer from './../';


jest.mock('axios');

describe('<CourseDetailsContainer>', () => {

  describe('when component mount', () => {
    let component;
    let container;
    let tree;

    it('should call and present a course', () => {

      const resp = { data: ResMock };
      component = <CourseDetailsContainer
        dispatchLoad={({ }) => { }}
        match={{ params: { id: 2 } }}
        selectedCourse={null}
        isLoadingCourse={false}
      />;

      (axios.get as any).mockResolvedValue(resp);

      container = (
        <Provider store={store}>
          <StaticRouter context={{}}>{component}</StaticRouter>
        </Provider>
      );

      tree = renderer.create(container).toJSON();

      expect(tree).toMatchSnapshot();
      expect(axios.get).toHaveBeenCalled();

    });


  });


});


const ResMock = {
  "id": 1,
  "name": "Física II",
  "description": "Curso remasterizado e criado inteiramente baseado na sua graduação da POLI USP. Provas passadas resolvidas, teoria explicadinha e muitos exercícios extra. Aproveite!",
  "generic": false,
  "highlight": true
}
