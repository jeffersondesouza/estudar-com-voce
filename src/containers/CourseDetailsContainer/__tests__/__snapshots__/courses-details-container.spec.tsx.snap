// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseDetailsContainer> when component mount should call and present a course 1`] = `
<div
  className="details"
>
  <div>
    <a
      className="details__back"
      href="/cocurses"
      onClick={[Function]}
    >
      ‚Üê
    </a>
  </div>
</div>
`;
